src/index.js

```diff
@@ -311,22 +328,34 @@ export const fuzzForInStatement = (f = new FuzzerState) => {
   return new Shift.ForInStatement({left, right, body});
 }
 
-export const fuzzForOfStatement = (f = new FuzzerState) => {
+function fuzzForOfParts(f) {
   f = f.goDeeper();
-  let left = f.rng.nextBoolean() ? fuzzVariableDeclaration(f, {inForInOfHead: true}) : fuzzAssignmentTarget(f);
+  let left;
+  do {
+    left = f.rng.nextBoolean() ? fuzzVariableDeclaration(f, {inForInOfHead: true}) : fuzzAssignmentTarget(f);
+    // https://github.com/tc39/ecma262/issues/2034
+  } while (left.type === 'AssignmentTargetIdentifier' && left.name === 'async');
```

Comment:
I don't love looping to avoid a random outcome. Can't we check for this case and replace it with something fixed?

Reply:
Works for me, done.